scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -34,21 +36,17 @@ class BeamWriter(val beamMode: BeamMode) extends Writer {
       case Some((coder, path)) =>
         val c: Coder[T] = coder.asInstanceOf[Coder[T]]
         Future(new Iterable[T] {
-          override def iterator: Iterator[T] = {
-            // Single dir by default just matches the dir, we need to match files inside
-            val matchedResources = FileSystems.`match`(s"$path*").metadata()
-            matchedResources.size() match {
-              case 0 => Iterator.empty
-              case 1 =>
-                val resource = matchedResources.get(0).resourceId()
-                val inputStream = Channels.newInputStream(FileSystems.open(resource))
+          // Single dir by default just matches the dir, we need to match files inside
+          val matchedResources = FileSystems.`match`(s"$path*").metadata().asScala
+
+          override def iterator: Iterator[T] =
+            matchedResources
+              .map { resource =>
```

Comment:
nit: I think `matchedResources.iterator.flatMap { ` clearer than manually reimplementing via fold.

Now that I think about this though, we are never closing the input stream.

I think we may need something like this:

```scala
def iterator = {
  // an empty Iterator that closes an InputStream when it is iterated
  def closeIt(is: InputStream): Iterator[T] =
    new Iterator[T] {
      def hasNext = {
        is.close()
        false
      }
      def next = Iterator.empty.next
    }

  matchedResources
    .iterator
    .flatMap { resource =>
      val is = Channels.newInputStream(FileSystems.open(resource.resourceId()))
      (new InputStreamIterator(is, c)) ++ closeIt(is)
    }
}
```
At least that would close the resource when you iterate through all the items.